lib/coinbase/address/wallet_address.rb

```diff
@@ -215,5 +245,44 @@ def broadcast_trade(trade, signed_payload:, approve_tx_signed_payload: nil)
 
       Coinbase::Trade.new(trade_model)
     end
+
+    def create_staking_operation(amount, asset_id, action, mode: :default, options: {})
+      operation_model = Coinbase.call_api do
+        asset = Coinbase::Asset.fetch(network_id, asset_id)
+        stake_api.create_staking_operation(
+          wallet_id,
+          id,
+          {
+            asset_id: asset.primary_denomination.to_s,
+            address_id: id,
+            action: action,
+            network_id: Coinbase.normalize_network(network_id),
+            options: {
+              amount: asset.to_atomic_amount(amount).to_i.to_s,
+              mode: mode
+            }.merge(options)
+          }
+        )
+      end
+
+      StakingOperation.new(operation_model)
+    end
+
+    def complete_staking_operation(amount, asset_id, action, mode: :default, options: {})
+      validate_staking_operation!(amount, asset_id, action, mode: mode, options: options)
+      op = create_staking_operation(amount, asset_id, action, mode: mode, options: options)
+      op.transactions.each_with_index do |transaction, i|
+        signed_payload = transaction.sign(@key)
```

Comment:
To avoid any extra errors would be good to broadcast only tx's that are signed and check that with the `signed?` function on the `Transaction` object.